grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.grpc.protocol;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.ClientDecoration;
+import com.linecorp.armeria.client.ClientOption;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.SimpleDecoratingClient;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.grpc.protocol.ArmeriaMessageDeframer.ByteBufOrStream;
+import com.linecorp.armeria.common.grpc.protocol.ArmeriaMessageDeframer.Listener;
+import com.linecorp.armeria.unsafe.ByteBufHttpData;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufHolder;
+import io.netty.buffer.Unpooled;
+
+/**
+ * A {@link UnaryGrpcClient} can be used to make requests to a gRPC server without depending on gRPC stubs.
+ * This client takes care of deframing and framing with the gRPC wire format and handling appropriate headers.
+ *
+ * <p>This client does not support compression. If you need support for compression, please consider using
+ * normal gRPC stubs or file a feature request.
+ */
+public class UnaryGrpcClient {
+
+    private final HttpClient httpClient;
+
+    /**
+     * Constructs a {@link UnaryGrpcClient} for the given {@link HttpClient}.
+     */
+    public UnaryGrpcClient(HttpClient httpClient) {
```

Comment:
I find this is not consistent with how other Armeria clients are instantiated. For example, I can imagine the following user code:

```java
UnaryGrpcClient client = new ClientBuilder("graw+https://...")
    ...
    .build(UnaryGrpcClient.class);

client.execute(path, payload);
```

Side note: Not an immediate goal, but a unary gRPC service/client could make use of `RpcRequest` and `RpcResponse`?

Reply:
Ah cool idea, will try it out since it doesn't look that tough!

I did think a bit about `RpcRequest/Response` but it seemed like it'd make the implementation quite complicated and I don't expect many users of the class so keeping it simple might be best.

Reply:
I realized it might not work well since we map `gproto` to the `application/grpc` media type since many gRPC upstream servers do not work with `grpc+proto`, at least at the time of writing that. `graw` would need to be the same content type, which I don't think we allow with `SerializationFormatProvider`. Do you think it's worth investigating it more?

Reply:
Ah, that's a problem. Let's add some TODO/FIXME and fix it later then. Thanks for giving a try!